src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -1193,10 +1192,12 @@ void addSubpatternsForTuple(ImmutableArray<TypeWithAnnotations> elementTypes)
         {
             Symbol? symbol = BindPropertyPatternMember(inputType, name, ref hasErrors, diagnostics);
 
-            if (inputType.IsErrorType() || hasErrors || symbol is null)
-                memberType = CreateErrorType();
-            else
-                memberType = symbol.GetTypeOrReturnType().Type;
+            memberType = symbol switch
+            {
+                FieldSymbol field => field.Type,
+                PropertySymbol property => property.Type,
+                _ => CreateErrorType()
```

Comment:
>CreateErrorType() [](start = 21, length = 17)

If this was causing a problem, is it still safe to use it for null result?

Reply:
> If this was causing a problem, is it still safe to use it for null result?

The problem was caused by an error type being used when the field or property type was available. In particular, the type returned from this method should match the type used for recursive property in `MakeTestsAndBindingsForRecursivePattern()`, which it should now.

---
In reply to: [597135186](https://github.com/dotnet/roslyn/pull/51950#discussion_r597135186) [](ancestors = 597135186)